samples/tacos/src/main/kotlin/com/slack/circuit/tacos/step/FillingsOrderStep.kt

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2023 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.tacos.step
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.RadioButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.produceState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import com.slack.circuit.tacos.OrderDetails
+import com.slack.circuit.tacos.R
+import com.slack.circuit.tacos.model.Ingredient
+import com.slack.circuit.tacos.repository.IngredientsRepository
+import kotlinx.collections.immutable.ImmutableList
+
+object FillingsOrderStep : OrderStep {
+  override val index = 0
+  override val headerResId = R.string.fillings_step_header
+
+  sealed interface State : OrderStep.State {
+    object Loading : State
+    data class AvailableFillings(
+      val selected: Ingredient? = null,
+      val list: ImmutableList<Ingredient>,
+      val eventSink: (Event) -> Unit
+    ) : State
+  }
+
+  sealed interface Event {
+    data class SelectFilling(val ingredient: Ingredient) : Event
+  }
+}
+
+internal class FillingsProducerImpl(private val repository: IngredientsRepository) :
+  OrderStep.StateProducer<FillingsOrderStep.State> {
```

Comment:
Continuing from my comment higher up - this definitely wants to be a `Presenter`

Reply:
https://github.com/slackhq/circuit/pull/496#discussion_r1149871408